mlir/lib/Dialect/Bufferization/IR/BufferizableOpInterface.cpp

```diff
@@ -961,6 +961,16 @@ FailureOr<BufferLikeType> bufferization::detail::defaultGetBufferType(
   assert(llvm::isa<TensorType>(value.getType()) && "expected tensor type");
   auto tensorType = cast<TensorType>(value.getType());
 
+  auto elementType = tensorType.getElementType();
+
+  if (!BaseMemRefType::isValidElementType(elementType)) {
+    if (Operation *def = value.getDefiningOp())
```

Comment:
Use `getOwnerOfValue(value)->emitError()` to make sure that we print an error message even for block arguments.